### PR TITLE
Add support for `SwapGate`, `iSwapGate`, and `DCXGate`

### PR DESCRIPTION
**Pull request by garrison**
I might clean this up a bit, but hey -- it works. :slightly_smiling_face: 

#### Remaining action items

- [x] fix coverage (one line is currently missing)
- [x] clean up & add comments
- [x] add to release note (might need to rebase after #292 is merged to please reno)
- [x] possibly add additional tests of `kappa` values from other recent gates
- [x] update supported gates in docstring

### CODE REVIEW

**Comment by garrison** [comment]
This should get us 90% of the way to #186, as well.  I'll know for sure once I get tests to pass for arbitrary gates.

File: releasenotes/notes/additional-gates-f4ed6c0e8dc3a9be.yaml [hunk 1]
```diff
@@ -6,3 +6,5 @@ features:
     - :class:`~qiskit.circuit.library.CYGate`
     - :class:`~qiskit.circuit.library.SXGate`
     - :class:`~qiskit.circuit.library.ECRGate`
+    - :class:`~qiskit.circuit.library.SwapGate`
+    - :class:`~qiskit.circuit.library.iSwapGate`
```
Comment (garrison) [inline releasenotes/notes/additional-gates-f4ed6c0e8dc3a9be.yaml:10]: Add `DCXGate`

File: circuit_knitting/cutting/qpd/qpd.py [hunk 2]
```diff
@@ -236,6 +240,156 @@ def supported_gates() -> set[str]:
...
+    uu31 = u[3] * np.conj(u[1])
+    coeffs, maps1, maps2 = zip(
+        # First line of Eq. (19) in
+        # https://quantum-journal.org/papers/q-2021-01-28-388/
+        (np.abs(u[0]) ** 2, [], []),  # Identity
+        (np.abs(u[1]) ** 2, [XGate()], [XGate()]),
+        (np.abs(u[2]) ** 2, [YGate()], [YGate()]),
+        (np.abs(u[3]) ** 2, [ZGate()], [ZGate()]),
+        # Second line
+        (2 * np.real(uu01), A0x, A0x),
```
Comment (caleb-johnson) [inline circuit_knitting/cutting/qpd/qpd.py:309]: I'm a little confused why there are so many terms associated with lines 2 (and also 3) of eq 19. That summation should have 6 terms, and each of those terms looks like (uaua'* + ua'ua*)(Aaa' - Baa'). Can't quite figure out why there are 30 terms

Comment (caleb-johnson) [reply circuit_knitting/cutting/qpd/qpd.py:309]: Yes, I was just coming here to say I see that it's 6+12+12. Thanks! :)

**Review by caleb-johnson — APPROVED** [review]
This looks great. Thanks for adding the tests we missed in earlier PRs 👍

**Pull request merged**